website/events/api/v2/admin/views.py

```diff
@@ -0,0 +1,145 @@
+import json
+
+from django.http import HttpResponse, Http404
+from oauth2_provider.contrib.rest_framework import IsAuthenticatedOrTokenHasScope
+from rest_framework import status
+from rest_framework.generics import get_object_or_404
+from rest_framework.views import APIView
+from rest_framework import filters as framework_filters
+
+from events import services
+from events.api.v2.admin import filters
+from events.api.v2.admin.permissions import IsOrganiser
+from events.api.v2.admin.serializers.event import EventListSerializer, EventSerializer
+from events.api.v2.admin.serializers.event_registration import (
+    EventRegistrationSerializer,
+)
+from events.models import Event, EventRegistration
+from thaliawebsite.api.v2.admin.views import (
+    AdminListAPIView,
+    AdminRetrieveAPIView,
+    AdminCreateAPIView,
+    AdminUpdateAPIView,
+    AdminDestroyAPIView,
+)
+import events.api.v2.filters as normal_filters
+
+
+class EventAdminListAPIView(AdminListAPIView):
+    queryset = Event.objects.prefetch_related("organiser")
+    serializer_class = EventListSerializer
+    permission_classes = [IsAuthenticatedOrTokenHasScope]
+    required_scopes = ["events:admin"]
+    filter_backends = [
+        framework_filters.OrderingFilter,
+        normal_filters.CategoryFilter,
+        normal_filters.OrganiserFilter,
+        normal_filters.EventDateFilter,
+        filters.PublishedFilter,
+    ]
+    ordering_fields = (
+        "start",
+        "end",
+        "published",
+        "registration_start",
+        "registration_end",
+    )
+
+
+class EventAdminDetailAPIView(
+    AdminRetrieveAPIView, AdminCreateAPIView, AdminUpdateAPIView, AdminDestroyAPIView
```

Comment:
I think the DetailAPIView shouldn't be a CreateAPIView, only UpdateAPIView. The ListAPIView should be a CreateAPIView in my opinion,

Reply:
That would overcomplicate the other view, which shows a list. This one is for single objects and editing those. And it uses a different serializer. Thus I chose to include it here instead.

Reply:
But now, if you want to create a new event, you must do that via another event, also using the `id` of a different event.

I don't think it really makes the other view too complicated

Reply:
I was dumb